### DISCUSSION TITLE
Internal error function lexer failed

### DISCUSSION THREAD

**Discussion by Amuri44** [OpenModelica/OpenModelica#8509] [Q&A] [answered]
I'm trying to simulate a model that I made, but I got the below error:

[2] 17:23:29 Translation Error
[C:/OM118/OM64bit/OMCompiler/Compiler/NFFrontEnd/NFUnit.mo: 883:7-883:68]: Internal error function lexer failed

Anyway to solve it?

**Comment by casella** [comment, marked as answer]
@Amuri44 you should attach a test case to reproduce the issue, otherwise it's hard to figure out.

You may also try with the latest nightly build.

**Reply by Amuri44** [reply]
What is the latest nightly build? is it a version of OpenModelica?

**Comment by adrpo** [comment]
I guess you have a strange unit definition in your model. The compiler should give a warning and continue but it stops. The nightly build version is the latest build of OpenModelica from the master: 
https://build.openmodelica.org/omc/builds/windows/nightly-builds/

**Comment by casella** [comment]
I tried to run the model with the latest nightly build and I got this message
```
[4] 11:21:02 Translation Error
[M: 35:45-35:64]: Modified element limitsAtInit not found in class Limiter.
```
maybe if you fix that you will also be able to run it in v. 1.18.0

**Reply by perost** [reply]
> I tried to run the model with the latest nightly build and I got this message
> 
> ```
> [4] 11:21:02 Translation Error
> [M: 35:45-35:64]: Modified element limitsAtInit not found in class Limiter.
> ```
> 
> maybe if you fix that you will also be able to run it in v. 1.18.0

That's because you used MSL 4.0.0 while the model uses MSL 3.2.3.

**Reply by casella** [reply]
@perost, any idea why the lexer is not happy with MSL 3.2.3?

**Reply by perost** [reply]
> @perost, any idea why the lexer is not happy with MSL 3.2.3?

I would assume it has issues with the units in the given model, not in the MSL. But I'm no expert on units, so I can't really say what's going wrong there.

**Reply by casella** [reply]
In that case, it would be nice to have an error message that explicitly refers to the parsing of unit strings. A message like "Internal error function lexer failed" is not helping much, unless you have @adrpo's expertise, which is rather uncommon 😄 

Do you think that could be fixed easily?

**Comment by perost** [comment]
#8518 changes the internal error to an actual error message, so you will now instead get the (correct) error messages:
```
[PSS1A_Edit.mo:18:3-19:30:writable] Notification: Invalid unit expression ‘S^2‘.
[Modelica 3.2.3+maint.om/Blocks/Continuous.mo:396:5-396:54:writable] Warning: The following equation is INCONSISTENT due to specified unit information: firstOrder.T = T6
Warning: The units of following sub-expressions need to be equal:
- sub-expression "T6" has unit "S"
- sub-expression "firstOrder.T" has unit "s"
```
The first error is that `^` is not a valid character in a unit expression, if you want e.g. `s²` you just write `s2`. The second error is that you seem to be using `S` for seconds when it should actually be `s`, so the units don't match the ones in the MSL.
